File: bin/ui/src/scene/display.cc
// Copyright 2017 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "apps/mozart/src/scene/display.h"

#include <magenta/syscalls.h>

namespace mozart {
namespace scene {

Display::Display(uint32_t width, uint32_t height, float device_pixel_ratio)
    : first_vsync_(mx_time_get(MX_CLOCK_MONOTONIC)),
      width_(width),
      height_(height),
      device_pixel_ratio_(device_pixel_ratio) {}

uint64_t Display::GetLastVsyncTime() const {
  uint64_t current_time = mx_time_get(MX_CLOCK_MONOTONIC);
  uint64_t num_elapsed_intervals =
      (current_time - first_vsync_) / kHardcodedPresentationIntervalNanos;
  return first_vsync_ +
         num_elapsed_intervals * kHardcodedPresentationIntervalNanos;
}

uint64_t Display::GetVsyncInterval() const {
  return kHardcodedPresentationIntervalNanos;
}

}  // namespace scene
}  // namespace mozart

File: garnet/bin/setui/src/display.rs
// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

pub use self::display_fidl_handler::fidl_io;
pub use self::light_sensor_controller::LIGHT_SENSOR_SERVICE_NAME;

pub mod display_configuration;
pub mod display_controller;
mod display_fidl_handler;
mod light_sensor;
mod light_sensor_config;
pub mod light_sensor_controller;
pub mod types;

pub use light_sensor_config::LightSensorConfig;

#[cfg(test)]
pub(crate) use light_sensor::testing as light_sensor_testing;
